src/mprispropertiesadaptor_p.h

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (C) 2015-2023 Jolla Ltd.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef MPRISPROPERTIESADAPTOR_P_H
+#define MPRISPROPERTIESADAPTOR_P_H
+
+#include <QtCore/QObject>
+#include <QtDBus/QtDBus>
+
+namespace Amber {
+
+class MprisPlayerPrivate;
+
+/*
+ * Adaptor class for interface org.freedesktop.DBus.Properties
+ *
+ * This is needed because the standard QDBusAbstractAdaptor
+ * Properties implementation doesn't allow errors to be
+ * returned for the org.freedesktop.DBus.Properties.Set method,
+ * but these are needed buy MPRIS spec.
+ *
+ */
+class MprisPropertiesAdaptor: public QDBusAbstractAdaptor
+{
+    Q_OBJECT
+    Q_CLASSINFO("D-Bus Interface", "org.freedesktop.DBus.Properties")
+    Q_CLASSINFO("D-Bus Introspection", ""
+    "  <interface name=\"org.freedesktop.DBus.Properties\">\n"
+    "    <method name=\"Get\">\n"
+    "      <arg name=\"interface_name\" type=\"s\" direction=\"in\"/>\n"
+    "      <arg name=\"property_name\" type=\"s\" direction=\"in\"/>\n"
+    "      <arg name=\"value\" type=\"v\" direction=\"out\"/>\n"
+    "    </method>\n"
+    "    <method name=\"Set\">\n"
+    "      <arg name=\"interface_name\" type=\"s\" direction=\"in\"/>\n"
+    "      <arg name=\"property_name\" type=\"s\" direction=\"in\"/>\n"
+    "      <arg name=\"value\" type=\"v\" direction=\"in\"/>\n"
+    "    </method>\n"
+    "    <method name=\"GetAll\">\n"
+    "      <arg name=\"interface_name\" type=\"s\" direction=\"in\"/>\n"
+    "      <arg name=\"values\" type=\"a{sv}\" direction=\"out\"/>\n"
+    "      <annotation name=\"org.qtproject.QtDBus.QtTypeName.Out0\" value=\"QVariantMap\"/>\n"
```

Comment:
Not sure if annotations should be put here, but they shouldn't cause any harm either.

Reply:
You're right, adding the annotations here causes them to appear twice in the introspection. I've removed them from here to avoid that.